rcl_lifecycle/src/rcl_lifecycle.c

```diff
@@ -163,14 +165,14 @@ rcl_lifecycle_trigger_transition(
 
   // If we have a faulty transition pointer
   if (!transition) {
-    fprintf(stderr, "No transition found for node %s with key %d\n",
-      state_machine->current_state->label, key);
+    RCUTILS_LOG_ERROR("No transition found for node %s with key %d",
+      state_machine->current_state->label, key)
```

Comment:
The macro invocation should **not** be followed by a semicolon. The rational is that the macros might evaluate to *nothing* in case the macro (or at least the specific log level) is being compiled out for efficiency reasons. At that point a trailing semicolon would be left.